src/snowcli/cli/nativeapp/manager.py

```diff
@@ -304,16 +304,18 @@ def _create_dev_app(self, diff: DiffResult) -> None:
                         self.app_name, self.app_role, actual_owner
                     )
 
-                if not diff.has_changes():
-                    # the app already exists and is up-to-date
-                    # ensure debug_mode is up-to-date
+                if diff.has_changes():
+                    # the app needs to be upgraded
+                    log.info(f"Upgrading existing application {self.app_name}.")
                     self._execute_query(
-                        f"alter application {self.app_name} set debug_mode = {self.debug_mode}"
+                        f"alter application {self.app_name} upgrade using @{self.stage_fqn}"
```

Comment:
In addition to `using @stage` , should we keep `set debug_mode` from before to obey `# ensure debug_mode is up-to-date`?

Reply:
Can you do both in one command? We no longer `return`, so it falls through and sets the debug mode afterwards (see the changed test).